tools/azure-sdk-tools/ci_tools/conda/CondaConfiguration.py

```diff
@@ -0,0 +1,196 @@
+from typing import List, Any
+import os
+import bs4
+import urllib3
+from ci_tools.functions import str_to_bool
+
+http = urllib3.PoolManager()
+# arguments: |
+#       -c "${{ replace(convertToJson(parameters.CondaArtifacts), '"', '\"') }}"
+#       -w "$(Build.SourcesDirectory)/conda/conda-recipes"
+#
+# # Sample configuration yaml. This is converted to json before being passed a blob to the invoking script. (see directly above for conversion)
+#     - name: uamqp
+#         common_root: uamqp
+#         in_batch: true
+#         conda_py_versions:
+#         - "38"
+#         - "39"
+#         - "310"
+#         - "311"
+#         checkout:
+#         - package: uamqp
+#         download_uri: https://files.pythonhosted.org/packages/0b/d8/fc24d95e6f6c80851ae6738c78da081cd535c924b02c5a4928b108b9ed42/uamqp-1.6.5.tar.gz
+#   - name: azure-core
+#     common_root: azure
+#     in_batch: ${{ parameters.release_azure_core }}
+#     channels: "conda-forge"
+#     checkout:
+#     - package: azure-core
+#       version: 1.24.0
+#     - package: azure-core
+#       version: 1.24.0
+#   - name: azure-storage
+#     common_root: azure
+#     in_batch: ${{ parameters.release_azure_storage }}
+#     checkout:
+#     - package: azure-storage-blob
+#       checkout_path: sdk/storage
+#       version: 12.12.0
+#     - package: azure-storage-queue
+#       checkout_path: sdk/storage
+#       version: 12.3.0
+#     - package: azure-storage-file-share
+#       checkout_path: sdk/storage
+#       version: 12.8.0
+#     - package: azure-storage-file-datalake
+#       checkout_path: sdk/storage
+#       version: 12.7.0
+
+def get_pypi_page(package: str, version: str) -> bs4.BeautifulSoup:
+    url = f"https://pypi.org/project/{package}/{version}/"
+
+    try:
+        r = http.request("GET", url)
+    except Exception as e:
+        raise RuntimeError(f"We must get this the data {url} to continue proper assembly.")
```

Comment:
I'm not fully understanding this error message, but I could be missing context

Reply:
Also, would it be helpful to `raise from` the original exception here?

Reply:
> I'm not fully understanding this error message, but I could be missing context

Effectively, we use request.get to download the zipped sdist from pypi. If we can't get that data, we can't continue. Hard error. I'll be a bit more explicit.

Reply:
Ahhh, I see. In that case, I think a "from" before the URL would clear it up. Thanks!

Reply:
```python
        raise RuntimeError(f"This package \"{package}\" has been configured to retrieve package source distribution from PyPI at URL \"{url}\"." +
                           "Unable to retrieve these bits from PyPI. Check conda package configuration in conda-sdk-client.yml and retry.")
```